### PR TITLE
[release/5.0] Update ICU package

### PR DESCRIPTION
**Pull request by EgorBo**
Master PRs: https://github.com/dotnet/runtime/pull/40981 + https://github.com/dotnet/runtime/pull/41166 combined.

The newer ICU is smaller and is split into several files (groups of cultures, see https://github.com/dotnet/icu/pull/20).

/cc @marek-safar @lewing

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -16,6 +16,7 @@
     <add key="dotnet-eng" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json" />
     <add key="dotnet5" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json" />
     <add key="dotnet5-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5-transport/nuget/v3/index.json" />
+    <add key="dotnet6-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6-transport/nuget/v3/index.json" />
```
Comment (akoeplinger) [inline NuGet.config:19]: I don't think we can add this feed to release/5.0. We probably need to branch icu for 5.0 too and make it publish to dotnet5-transport.

Comment (EgorBo) [reply NuGet.config:19]: is this necessary?

Comment (marek-safar) [reply NuGet.config:19]: Yep, let's stay with dotnet5 feed

Comment (safern) [reply NuGet.config:19]: Yeah I agree we should branch off to release/5.0 and publish to the 5.0 channel on ICU. That way if we need to service ICU for 5.0 we do it there.

Comment (akoeplinger) [reply NuGet.config:19]: we created a branch for release/5.0 on dotnet/icu now.

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -4,9 +4,9 @@
       <Uri>https://github.com/dotnet/standard</Uri>
       <Sha>cfe95a23647c7de1fe1a349343115bd7720d6949</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.NETCore.Runtime.ICU.Transport" Version="5.0.0-preview.8.20410.1">
+    <Dependency Name="Microsoft.NETCore.Runtime.ICU.Transport" Version="6.0.0-alpha.1.20425.2">
```
Comment (marek-safar) [inline eng/Version.Details.xml:7]: Why 6.* ?

Comment (EgorBo) [reply eng/Version.Details.xml:7]: @akoeplinger any idea? it's the latest version of ICU in dotnet5 feed. and I don't see any "6.0" string in the repo

Comment (akoeplinger) [reply eng/Version.Details.xml:7]: @EgorBo that's the wrong package version, you want this one: https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet5-transport&package=Microsoft.NETCore.Runtime.ICU.Transport&protocolType=NuGet&version=5.0.0-rc.1.20425.3&view=overview

**Review by marek-safar — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

**Review by safern — APPROVED** [review]

**Pull request merged**